articles/basics-of-convolution-neural-networks/index.md

```diff
@@ -0,0 +1,168 @@
+
+![Stride, S](/engineering-education/basics-of-convolution-neural-networks/stride.PNG)<br>
+
+*[Image Source: Stanford](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-convolutional-neural-networks)*
+
+### What is Convolution Neural Networks (CNN)?
+
+Convolution Neural Networks (CNN) is a class of deep learning techniques popularly use to solve computer vision tasks.
+CNN learns the spatial features of inputs through a process known as backpropagation, as we will see in the image below. Backpropagation can be described as the backward propagation of errors. It is the core algorithm behind how neural networks learn. Learn more about it in this [video](/https://www.youtube.com/watch?v=Ilg3gGewQ5U/).
+
+Let's take a look at the architecture of a CNN.
+
+### Architecture Overview
+
+![Architecture of a CNN together with the training process](/engineering-education/basics-of-convolution-neural-networks/cnn-architecture.PNG)<br>
+
+*[Image Source: CrossMark](https://doi.org/10.1007/s13244-018-0639-9)*
+
+The CNN architecture consists of a stacking of three building blocks:
+
+1. Convolution layers
+
+2. Pooling layers, i.e., max pooling
+
+3. Fully connected (FC) layers
+
+#### Convolution layer
+
+A convolution layer is a key component of the CNN architecture. This layer helps us perform feature extractions on input data. This feature extraction is achieved using both convolution layers (linear operation) and activation functions (non-linear operation). Rectified Linear Unit (ReLU) is one of the most used activation functions in deep learning. 
+
+Let's touch a little bit on it.
+
+##### Rectified Linear Unit (ReLU)
+
+From the image above, we note that a convolution layer is combined with the ReLU layer. The purpose of the ReLU activation function is to introduce non-linearity into the network. Introducing non-linearity is important in deep learning because, in real life, almost all data is non-linear. ReLU returns a 0 value for all negative inputs, while for positive values, it returns that same value back (x).
+
+Mathematically, this is defined by: 
+
+f(x) = max (0, x)
+
+![Rectified Linear Unit](/engineering-education/basics-of-convolution-neural-networks/relu-function.PNG)<br>
+
+*[Image Source: Papers with code](https://paperswithcode.com/method/relu)*
+
+ReLU is the default activation function when working with CNNs, as it tends to have a faster training time. 
+
+Other activation functions used in neural networks include the sigmoid and the hyperbolic tangent. Read more about them [here](/https://en.wikipedia.org/wiki/Activation_function/).
+
+#### Pooling layer
+
+A pooling layer is a form of non-linear downsampling that reduces the dimensionality of the feature maps of an input image, making its representation smaller and more manageable. This process reduces the number of parameters and computation resources in the network.
+
+It is important to note that a 2 x 2 filter size applied with a stride of 2 is the most used in CNN.
```

Comment:
Take an example and explain the same. What are the initial dimensions, and how they are reduced using pooling operation?